applications/nrf_desktop/doc/porting_guide.rst

```diff
@@ -3,8 +3,116 @@
 Application porting guide
 #########################
 
+The following document describes configuration and provides a porting guide for the nRF Desktop project. 
+
+Overview
+********
+
+Configuration sources
+=====================
+
+The nRF Desktop project uses following files as a sources of its configuration:
+
+    * DTS files - these reflect HW configuration (:ref:`device-tree`).
+    * Kconfig files - these reflect SW configuration (:ref:`kconfig_tips_and_tricks`).
+    * _def files - specific to the nRF Desktop - contain configuration arrays for the application modules.
+
+
+Board configuration
+===================
+
+The nRF Desktop application is modular and depending on a requested functions can run provide a mouse, keyboard or dongle functionality.
+The selected set of modules depends on the chosen role and also on the selected board.
+
+For a board to be supported by the application a set of configuration files must be provided for in the directory ``applications/nrf_desktop/configuration/your_board_name``.
+Inside this directory there should reside:
+    * Application configuration file for ``ZDebug`` build type (mandatory).
+    * Application configuration files for other build types (optional).
+    * Configuration file for the bootloader (optional).
+    * Configuration files for the selected modules (mandatory).
+    * DTS overlay file (optional).
+
+The application configuration files define a set of options with which the nRF Desktop project will be created for this board and selected build type.
+The nRF Desktop comes with configuration files for following boards:
+    * nrf52840_pca20041 - gaming mouse.
+      The board used is defined in ``nrf/boards/arm/nrf52840_pca20041`` for project specific HW.
+      The application is configured to act as a mouse with both BLE and USB transports enabled as input.
```

Comment:
`as output` (of the mouse) or `for user input`

Reply:
fixed